### ISSUE TITLE
Help users accept Azure VM image license prior to deploying Azure clusters

### ISSUE THREAD

**Issue by karuppiah7890** [vmware-tanzu/tanzu-framework#1223]
**Describe the feature request**

Currently users are asked to accept the Azure VM image license prior to deploying Azure clusters. For example, look at TCE docs for planning for Azure cluster deployment - https://tanzucommunityedition.io/docs/latest/azure-mgmt/#accept-the-base-image-license . Currently users have to manually accept this image license. It would be nice if this documentation is converted to a step in TKG CLI and Kickstart UI where the user is asked if they would like to accept the Azure VM image license and then they can accept it. This way, it's a smooth experience for the user. I like the idea of converting website docs to code docs / interactive and automatic explicit input from user, where users don't have to read the docs or know a lot of things just before getting started and instead can figure out on the go if the tool / services helps with it

I would like Tanzu Framework to improve the experience here for all the users, especially new users, who are new to Azure or DevOps but know enough to look around and get unblocked. I think that currently all users just have to read the docs a lot before even getting started and need a lot of knowledge to understand what's going on. I would like to reduce that slightly maybe, by providing assistance at Tanzu Framework level, just assistance, to new folks or experts in the domain

**Describe alternatives you've considered**

I haven't thought about any alternatives / workarounds, apart from leaving it as is - what's already happening - users manually accepting VM image license

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [X] Kickstart UI

**Additional context**

Related issues that lead to think about this issues - https://github.com/vmware-tanzu/community-edition/issues/2574 , https://github.com/vmware-tanzu/community-edition/issues/2586

**Comment by tenczar** [comment]
It might make sense to provide reference to this azure cli command: https://docs.microsoft.com/en-us/cli/azure/vm/image/terms?view=azure-cli-latest If we can add a validator that checks for the acceptance of the terms through the Azure API and then tells a user to accept the terms using the azure cli that might make sense. We need to take great care to not accept the terms for a user without making it explicitly clear that that is being done.

**Comment by karuppiah7890** [comment]
@tenczar makes sense! Surely we don't want to accept the license on behalf of the user without the user knowing that they are doing it. I was just trying to think - Can we show the license and ask the user to agree to it from our CLI/UI ? And about the checking of acceptance of the terms through Azure API - sure thing, that would be helpful too as a first step, so that users don't see errors later that image license was not accepted
